openmcyclus/depletion.py

```diff
@@ -0,0 +1,155 @@
+import numpy as np
+import openmc
+import openmc.deplete as od
+import os
+from xml.dom import minidom
+
+
+class Depletion(object):
+    def __init__(self, path, prototype, chain_file, timesteps, power):
+        '''
+        Class to hold objects related to calling the IndependentOperator
+        in OpenMC to perform stand alone depletion.
+
+        Inputs:
+        ----------
+        path: str
+            path to file for OpenMC model (geometry, materials,
+            and settings) and the results (depletion_results.h5)
+        prototype: str
+            name of prototype undergoing depletion
+        chain_file: str
+            name of file with decay chain data. Just the file name is
+            required, it is assumed that the file is in the same
+            location as the other files for the OpenMC model
+        timesteps: int
+            number of time steps to perform irradiation. It is assumed that
+            the number given is in units of months.
+        power: int
+            power output of the reactor, assumed in MWe.
+        '''
+        self.path = path
+        self.prototype = prototype
+        self.chain_file = chain_file
+        self.timesteps = timesteps
+        self.power = power
+
+    def read_model(self):
+        '''
+        Read the .xml files in the defined path to create an OpenMC
+        model in the python API. The files are assumed to be named
+        geometry.xml, materials.xml, settings.xml, and (optionally)
+        tallies.xml
+
+        Returns:
+        ---------
+        model: openmc.model.model object
+            OpenMC model of reactor geometry.
+        '''
+        model = openmc.Model.from_xml(
+            geometry=str(
+                self.path +
+                "geometry.xml"),
+            materials=str(
+                self.path +
+                "materials.xml"),
+            settings=str(
+                self.path +
+                "materials.xml"))
+        return model
```

Comment:
This allows optional support for `tallies.xml`
```suggestion
        model_kwargs = {"geometry": (self.path / "geometry.xml"),
                        "materials": (self.path / "materials.xml"),
                        "settings": (self.path / "settings.xml")}
        tallies_path = self.path / "tallies.xml"
        if tallies_path.exists():
            model_kwargs["tallies"] = tallies_path
        model = openmc.Model.from_xml(**model_kwargs)
        return model
```